pkg/cmd/server/origin/master.go

```diff
@@ -180,20 +180,24 @@ func (c *MasterConfig) newAssetServerHandler() (http.Handler, error) {
 	return assetServer.GenericAPIServer.PrepareRun().GenericAPIServer.Handler.FullHandlerChain, nil
 }
 
-func (c *MasterConfig) newOAuthServerHandler() (http.Handler, error) {
+func (c *MasterConfig) newOAuthServerHandler() (http.Handler, map[string]apiserver.PostStartHookFunc, error) {
 	if c.Options.OAuthConfig == nil {
-		return http.NotFoundHandler(), nil
+		return http.NotFoundHandler(), nil, nil
 	}
 
 	config, err := NewOAuthServerConfigFromMasterConfig(c)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	oauthServer, err := config.Complete().New(apiserver.EmptyDelegate)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return oauthServer.GenericAPIServer.PrepareRun().GenericAPIServer.Handler.FullHandlerChain, nil
+	return oauthServer.GenericAPIServer.PrepareRun().GenericAPIServer.Handler.FullHandlerChain,
+		map[string]apiserver.PostStartHookFunc{
+			"oauth.openshift.io-RegisterTokenRequestEndpoint": config.RegisterTokenRequestEndpoint,
```

Comment:
> I wasn't expecting handler registration in post-start hooks... what requires this to be async?

The creation of clients needs to be a post-start hook, the handler could give failures until they exist.